controllers/etcdcluster_controller.go

```diff
@@ -162,6 +162,41 @@ func (r *EtcdClusterReconciler) getStorageOSClientSecret(ctx context.Context, se
 	return secret, nil
 }
 
+// getDefaultStorage returns the the default storage class in the cluster, if more than one storage
+// class is set to default, the first one discovered is returned. An error is returned if no default
+// storage class is found.
+func (r *EtcdClusterReconciler) getDefaultStorageClass(ctx context.Context) (*kstoragev1.StorageClass, error) {
+	storageClasses := &kstoragev1.StorageClassList{}
+	err := r.List(ctx, storageClasses)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// No storageclass found
+			return nil, nil
```

Comment:
I think based on this comment `An error is returned if no default storage class is found`  we have to return an error.

Reply:
Here it looks like it's when no storageclass is found at all, not that we didn't find a default one. That happens below at the end of this function

Reply:
@Ricardo-Osorio I see, but the comment tells something different. /that's why i never trust in comments and usually try to avoid/. `apierrors.IsNotFound(err)` == `no default storage class is found`, so i expect `An error is returned`. We have to:
a) change the comment
b) change the behavior

Reply:
After looking at the ticket again: `the etcdcluster CR requires an explicit storageClassName for the volume. `
Therefore if it has not been set **and** we cannot find a storage class to use, then returning an error makes sense here imo.